src/CommandAndStateMessageParser.cpp

```diff
@@ -0,0 +1,317 @@
+}
+
+TiltCameraHead CommandAndStateMessageParser::getCameraHeadInfo(string address)
+{
+    TiltCameraHead camera_head;
+    camera_head.light = mJData["devices"][address]["cameraHead"]["lights"].asDouble();
+    camera_head.laser = mJData["devices"][address]["cameraHead"]["lasers"].asBool();
+    vector<float> tilt_command;
+    tilt_command.push_back(
+        mJData["devices"][address]["cameraHead"]["tilt"]["speed"].asFloat()
+    );
+    camera_head.tilt_command = samples::Joints::Speeds(tilt_command);
+    MotorDiagnostics motor;
+    motor.current =
+        mJData["devices"][address]["cameraHead"]["tiltMotorDiagnostics"]["current"]
+            .asDouble();
+    motor.overcurrent =
+        mJData["devices"][address]["cameraHead"]["tiltMotorDiagnostics"]["overcurrent"]
+            .asBool();
+    vector<float> vector_motor;
+    vector_motor.push_back(
+        mJData["devices"][address]["cameraHead"]["tiltMotorDiagnostics"]["rpm"].asFloat()
+    );
+    motor.motor = samples::Joints::Raw(vector_motor);
+    camera_head.tilt_motor_diagnostics = motor;
+    camera_head.camera.brightness =
+        mJData["devices"][address]["cameraHead"]["camera"]["brightness"].asDouble();
+    camera_head.camera.exposure =
+        mJData["devices"][address]["cameraHead"]["camera"]["exposure"]["value"].asDouble(
+        );
+    camera_head.camera.focus =
+        mJData["devices"][address]["cameraHead"]["camera"]["focus"]["value"].asDouble();
+    camera_head.camera.saturation =
+        mJData["devices"][address]["cameraHead"]["camera"]["saturation"].asDouble();
+    camera_head.camera.sharpness =
+        mJData["devices"][address]["cameraHead"]["camera"]["sharpness"].asDouble();
+    camera_head.camera.zoom.ratio =
+        mJData["devices"][address]["cameraHead"]["camera"]["zoom"]["ratio"].asDouble();
+    vector<float> vector_speed;
+    vector_speed.push_back(
+        mJData["devices"][address]["cameraHead"]["camera"]["zoom"]["speed"].asFloat()
+    );
+    camera_head.camera.zoom.speed = samples::Joints::Speeds(vector_speed);
+
+    return camera_head;
+}
+
+double CommandAndStateMessageParser::getLightInfo(string address)
+{
+    return mJData["devices"][address]["auxLights"].asDouble();
+}
+
+double CommandAndStateMessageParser::getTetherDistanceInfo(string address)
+{
+    return mJData["devices"][address]["distance"].asDouble();
+}
+
+double CommandAndStateMessageParser::getTemperatureInfo(string address)
+{
+    return mJData["devices"][address]["cpuTemp"].asDouble();
+}
+
+bool CommandAndStateMessageParser::getCalibrateInfo(string address)
+{
+    return mJData["devices"][address]["calibrator"].asBool();
+}
+
+bool CommandAndStateMessageParser::getReadyInfo(string address)
+{
+    return mJData["devices"][address]["ready"].asBool();
+}
+
+bool CommandAndStateMessageParser::getLeakInfo(string address)
+{
+    return mJData["devices"][address]["leak"].asBool();
+}
+
+bool CommandAndStateMessageParser::getACPowerInfo(string address)
```

Comment:
`isACPowerConnected`. I'll stop here. Just change all the `*Info` method names to something that actually describes what is being returned.